sdk/src/main/scala/com/horizen/SidechainApp.scala

```diff
@@ -289,15 +279,11 @@ class SidechainApp @Inject()
     storage
   }
 
-  override def configure(binder: Binder): Unit = {
-    binder.bind(classOf[SidechainBoxesDataCompanion])
-      .toInstance(sidechainBoxesDataCompanion)
-
-    binder.bind(classOf[SidechainProofsCompanion])
-      .toInstance(sidechainProofsCompanion)
-
-    binder.install(new FactoryModuleBuilder()
-      .build(classOf[SidechainCoreTransactionFactory]))
+  def getTransactionActorRef(): ActorRef = {
+    return sidechainTransactionActorRef
+  }
+  def getNodeViewActorRef(): ActorRef = {
+    return nodeViewHolderRef
```

Comment:
I'm a bit worried, that we expose the NodeViewHolderRef outside the SidechainApp. Why not to return an instance of NodeViewHelper directly?

Reply:
if we return it directly, we would not be able to inject directly NodeViewHelper in the app because we would have a circular dependency (sidechainpp require our app module to start that in turns would require it to access to the viewhelper)

Reply:
Ok. So can we return some wrapper on top of `NodeViewHolderRef` then? Like a NodeViewProvider with the "withNodeView` method, for example.

Why I'm talking about this, is that giving direct access to the `NodeViewHolderRef` outside the SidechainApp is strictly forbidden, because will give a direct access to the SidechainHistory/State/Mempool/Wallet that have modification methods. The usage of which will broke the node.

Reply:
yes, added the wrappers